app/models/concerns/client_file_base.rb

```diff
@@ -16,7 +16,8 @@ module ClientFileBase
 
     has_one_attached :client_file
 
-    validates_presence_of :name
+    # Name is really always required, this conditional just makes the client_file validation take precedence.
+    validates_presence_of :name, if: -> { client_file.attached? }
```

Comment:
I think maybe the comment is just confusing me here, what do you mean by "client_file validation"

It looks like the `file_exists_and_not_too_large` validation is currently validating that the `client_file` exists,  so we can rely on that and only validate presence of `name` if the file IF at has a `client_file` attachment.. is that right?

Reply:
Adding some screenshots from testing for reference:


This is what I see if I try to submit the File form without actually uploading anything. The first error is coming from the Form Definition validation code, because the "Attachment" field is marked as `required: true`. The second error is coming from here. (The issue of duplicated errors here pre-dates this PR).
<img width="920" alt="Screenshot 2024-12-09 at 9 39 44 AM" src="https://github.com/user-attachments/assets/3629384b-2573-4005-b58d-9f6ba92e4317">

This is what I see if I try to upload a blob that has been expired:
(The form required-ness validation isn't present because there was a blob id submitted)

<img width="900" alt="Screenshot 2024-12-09 at 11 23 42 AM" src="https://github.com/user-attachments/assets/902289a9-b558-4ee9-bc81-ee266db6dc75">

Reply:
I'm sorry about the confusing comment. Yes, your summary is right, so I'll update the comment to match (if you want to keep this, that is). I added this conditional in order to get rid of "Name can't be blank" in the screenshot below.
![Screenshot 2024-12-09 at 11 57 57 AM](https://github.com/user-attachments/assets/71da1273-7793-4339-8417-b5065f83ebc4)

Reply:
I see, yes it makes sense to me to keep your change. Thanks!